src/network-uptime/network-dashboard.py

```diff
@@ -0,0 +1,154 @@
+import os
+from dotenv import load_dotenv
+import numpy as np
+import pandas as pd
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.ticker import MaxNLocator
+import requests
+import streamlit as st
+
+dotenv_path = os.path.join(os.path.dirname(__file__), '.env')
+load_dotenv(dotenv_path)
+
+BASE_API_URL = os.getenv('BASE_URL')
+
+matplotlib.use('Agg')
+
+st.set_page_config(page_title='NETWORK UPTIME AND DOWNTIME', layout="wide")
+st.title('Data report on network uptime and outages over the first quarter, 2022')
+
+
+api_url = f"{BASE_API_URL}/devices?tenant=airqo"
+results = requests.get(api_url)
+devices_data = results.json()["devices"]
+
+device_name = devices_name = [data['name'] for data in devices_data]
+selected_device = st.sidebar.selectbox('Device name', device_name)
+
+
+@st.cache
+def load_device_data(device):
+    api = f"{BASE_API_URL}/monitor/devices/uptime?tenant=airqo&startDate=2022-01-01T00:00:00.001Z&endDate=2022-03-25T19:00:00.001Z&device_name=' + device
+    api_result = requests.get(api)
+    values = api_result.json()['data'][0]['values']
+    subset = ['created_at', 'uptime', 'downtime']
+    data_subset_list = []
+    for element in values:
+        data_subset = {}
+        for k in subset:
+            data_subset[k] = element[k]
+        data_subset_list.append(data_subset)
+
+    return data_subset_list
+
+
+def load_device_dataframe(json_data):
+    device_data = pd.DataFrame(json_data)
+    device_data['created_at'] = pd.to_datetime(
+        device_data['created_at']).dt.date
+
+    device_data.set_index('created_at', inplace=True)
+
+    device_uptime_data = device_data.loc[(
+        device_data.uptime == 100)]
+    device_downtime_data = device_data.loc[(
+        device_data.uptime == 0)]
+
+    device_uptime_count = pd.DataFrame(device_uptime_data.groupby(
+        pd.DatetimeIndex(device_uptime_data.index).date).count()['uptime'])
+    device_downtime_count = pd.DataFrame(device_downtime_data.groupby(
+        pd.DatetimeIndex(device_downtime_data.index).date).count()['downtime'])
+
+    device_name_data = device_uptime_count.join(
+        device_downtime_count, how='outer').fillna(0).astype(np.int32)
+    #device_name_data = device_name_data.add_suffix('_count')
+    device_name_data.index = pd.to_datetime(device_name_data.index)
+    device_name_data['pct_uptime'] = (
+        device_name_data['uptime']/device_name_data.sum(axis=1)*100).round(1)
+    device_name_data['pct_downtime'] = 100 - device_name_data['pct_uptime']
+
+    return device_name_data
+
```

Comment:
This function build a dataframe, a tabular representation of the device data. This table contains the following four columns: uptime counts, downtime counts, uptime percent and downtime percent all in hourly frequency.